src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -1929,6 +1929,9 @@ internal static unsafe void NumberToStringFormat(ref ValueStringBuilder sb, ref
                     }
                 }
             }
+
+            if (number.sign && (section == 0) && (number.scale == 0) && (sb.Length > 0))
+                sb.Insert(0, info.NegativeSign);
```

Comment:
Does this have any measurably negative impact on perf?  Inserting into a StringBuilder isn't particularly efficient.